x/evm/precompile/erc20/proxy/ERC20Proxy.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract ERC20Proxy {
+    // Address of the precompiled ERC20 contract
+    address public constant ERC20_PRECOMPILED_ADDRESS = address(0x0000000000000000000000000000000000000101);
+
+    // Event for successful registration
+    event RegistrationSuccess(bool success);
+
+    // ERC20 standard events
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+
+    constructor(string memory denom) {
+        bool success = register(denom);
+        emit RegistrationSuccess(success);
+    }
+
+    /**
+     * @dev Fallback function that forwards all unknown calls to the precompiled ERC20 contract.
+     */
+    fallback() external payable {
+        _delegate(ERC20_PRECOMPILED_ADDRESS);
+    }
+
+    /**
+     * @dev Receive function to accept plain Ether transfers.
+     */
+    receive() external payable {}
+
+    /**
+     * @dev Delegates the request to the target address and returns the result to the caller.
+     * @param target The address of the target contract (i.e., the precompiled ERC20 contract).
+     */
+    function _delegate(address target) internal {
+        (bool success, bytes memory returndata) = target.delegatecall(msg.data);
+        assembly {
+            switch success
+            case 0 { revert(add(returndata, 32), mload(returndata)) } // Revert on failure
+            default { return(add(returndata, 32), mload(returndata)) } // Return data on success
+        }
+    }
+
+    /**
+     * @dev Registers the token with the specified denom in the precompiled ERC20 contract.
+     * @param denom The token denomination to register.
+     * @return success Whether the registration was successful.
+     */
+    function register(string memory denom) internal returns (bool) {
```

Comment:
Two issues with this method:
1. Why the type of denom is `string`, what information is packed in this string?
2. Is there any permission control needed for this method? It seems that this should be an admin method to me.

Reply:
1. The denom represents the underlying IBC asset, similar to `ibc/xxx`. If optimization is needed, it can be represented in a format like` {“eth”, "channel0"}`. Improvements to enhance this experience may be introduced later if required.

2. Based on the design, the ERC20 proxy does not restrict registration permissions. It merely serves as an operational interface for IBC assets. For asset transfers, we only verify that the sender matches the signer. Please refer to the design details at: https://forum.artela.network/t/detailed-desgined-of-ibc-erc20/13#p-15-h-41-token-pair-registration-8.